backend/infrahub/core/relationship/model.py

```diff
@@ -1179,6 +1179,8 @@ async def save(
                         old_data=details.peers_database[rel.peer_id],
                         properties_to_update=properties_not_matching,
                     )
+            elif rel.schema.kind == RelationshipKind.PARENT:
+                relationship_mapper.add_parent_from_relationship(relationship=rel)
```

Comment:
Why do we need some logic here in addition to what's in `Node._update`?

Reply:
The stuff in Node._update is only for when the relationship wasn't touched at all in the update but we still need to query for the relationship of the parent.